src/Components/Components/src/Rendering/InvalidEventException.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Components.Rendering
+{
+    /// <summary>
+    /// Thrown when the renderer receives an invalid event ID that it can't dispatch.
+    /// </summary>
+    public class InvalidEventException : Exception
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="InvalidEventException"/>.
+        /// </summary>
+        /// <param name="eventId">The id of the invalid event.</param>
+        public InvalidEventException(ulong eventId)
+            : this(eventId, $"There is no event handler associated with this event.")
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="InvalidEventException"/>.
+        /// </summary>
+        /// <param name="eventId">The id of the invalid event.</param>
+        /// <param name="message">The message explaining the reason for the error.</param>
+        /// <param name="exception">The original exception that caused the issue.</param>
+        public InvalidEventException(ulong eventId, string message, Exception exception) : base(message + $" EventId: {eventId}", exception)
```

Comment:
Should we order these the same way the base class does?

Reply:
`message + $" EventId: {eventId}"` do we need this? 

Usually it's pretty surprising when a constructor modifies the value you give it. Like if a class has a message property and a message constructor parameter, I don't want anything tricky to happen 😆 

Should we just inline this into the places where we need it?

Reply:
My intention was to try and ensure that the event ID is always registered when we log/view the exception. We definitely could inline it in the places where we used. 

I was not sure about providing structured data vs ensuring the message always contains all the information.

Reply:
I'm going to take it out and simply leave the string and string,exception overloads.

Reply:
done